## View the file

In [1]:
FDIRY=/hpc/group/igvf/K562/mdl_eval/BlueSTARR_vs_DeltaSVM/Kircher_entire_regions
FNAME=Kircher_region_chunk.txt
FPATH=${FDIRY}/${FNAME}
cat ${FPATH} | head -n 1

chr11:5227021-5227208	5227021:ref=G:G,C,A,T	5227022:ref=G:G,C,A,T	5227023:ref=T:G,C,A,T	5227024:ref=G:G,C,A,T	5227025:ref=T:G,C,A,T	5227026:ref=C:G,C,A,T	5227027:ref=T:G,C,A,T	5227028:ref=G:G,C,A,T	5227029:ref=T:G,C,A,T	5227030:ref=T:G,C,A,T	5227031:ref=T:G,C,A,T	5227032:ref=G:G,C,A,T	5227033:ref=A:G,C,A,T	5227034:ref=G:G,C,A,T	5227035:ref=G:G,C,A,T	5227036:ref=T:G,C,A,T	5227037:ref=T:G,C,A,T	5227038:ref=G:G,C,A,T	5227039:ref=C:G,C,A,T	5227040:ref=T:G,C,A,T	5227041:ref=A:G,C,A,T	5227042:ref=G:G,C,A,T	5227043:ref=T:G,C,A,T	5227044:ref=G:G,C,A,T	5227045:ref=A:G,C,A,T	5227046:ref=A:G,C,A,T	5227047:ref=C:G,C,A,T	5227048:ref=A:G,C,A,T	5227049:ref=C:G,C,A,T	5227050:ref=A:G,C,A,T	5227051:ref=G:G,C,A,T	5227052:ref=T:G,C,A,T	5227053:ref=T:G,C,A,T	5227054:ref=G:G,C,A,T	5227055:ref=T:G,C,A,T	5227056:ref=G:G,C,A,T	5227057:ref=T:G,C,A,T	5227058:ref=C:G,C,A,T	5227059:ref=A:G,C,A,T	5227060:ref=G:G,C,A,T	5227061:ref=A:G,C,A,T	5227062:ref=A:G,C,A,T	5227063:ref=G:G,C,A,T	5227064:ref=C:G,C,A,T	5227065:re

In [2]:
FDIRY=/hpc/group/igvf/K562/mdl_eval/BlueSTARR_vs_DeltaSVM/Kircher_entire_regions
FNAME=Kircher_region_chunk.txt
FPATH=${FDIRY}/${FNAME}
cat ${FPATH} | head -n 2 | tail -n 1

chr11:5249804-5250078	5249804:ref=G:G,C,A,T	5249805:ref=G:G,C,A,T	5249806:ref=C:G,C,A,T	5249807:ref=G:G,C,A,T	5249808:ref=T:G,C,A,T	5249809:ref=C:G,C,A,T	5249810:ref=T:G,C,A,T	5249811:ref=G:G,C,A,T	5249812:ref=G:G,C,A,T	5249813:ref=A:G,C,A,T	5249814:ref=C:G,C,A,T	5249815:ref=T:G,C,A,T	5249816:ref=A:G,C,A,T	5249817:ref=G:G,C,A,T	5249818:ref=G:G,C,A,T	5249819:ref=A:G,C,A,T	5249820:ref=G:G,C,A,T	5249821:ref=C:G,C,A,T	5249822:ref=T:G,C,A,T	5249823:ref=T:G,C,A,T	5249824:ref=A:G,C,A,T	5249825:ref=T:G,C,A,T	5249826:ref=T:G,C,A,T	5249827:ref=G:G,C,A,T	5249828:ref=A:G,C,A,T	5249829:ref=T:G,C,A,T	5249830:ref=A:G,C,A,T	5249831:ref=A:G,C,A,T	5249832:ref=C:G,C,A,T	5249833:ref=C:G,C,A,T	5249834:ref=T:G,C,A,T	5249835:ref=C:G,C,A,T	5249836:ref=A:G,C,A,T	5249837:ref=G:G,C,A,T	5249838:ref=A:G,C,A,T	5249839:ref=C:G,C,A,T	5249840:ref=G:G,C,A,T	5249841:ref=T:G,C,A,T	5249842:ref=T:G,C,A,T	5249843:ref=C:G,C,A,T	5249844:ref=C:G,C,A,T	5249845:ref=A:G,C,A,T	5249846:ref=G:G,C,A,T	5249847:ref=A:G,C,A,T	5249848:re

## Original data

- Data
    - https://kircherlab.bihealth.org/satMutMPRA/
    - https://doi.org/10.17605/OSF.IO/75B2M
- Reference
    - https://www.nature.com/articles/s41467-019-11526-w
    - https://github.com/kircherlab/MPRA_SaturationMutagenesis
    - https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE126550

In [3]:
ls -l /hpc/group/igvf/benchmarks/Kircher_et_al_2019

total 4565
drwxr-xr-x. 2 42834 majoroslab    502 Mar 20 13:44 Enhancers
-rw-r--r--. 1 42834 majoroslab   1713 Mar 20 13:34 Enhancers.txt
-rw-r--r--. 1 42834 majoroslab 111164 Mar  6 21:08 GRCh38_BCL11A.tsv
-rw-r--r--. 1 42834 majoroslab  53410 Mar  6 21:08 GRCh38_F9.tsv
-rw-r--r--. 1 42834 majoroslab 108702 Mar  6 21:08 GRCh38_FOXE1.tsv
-rw-r--r--. 1 42834 majoroslab  71664 Mar  6 21:08 GRCh38_GP1BA.tsv
-rw-r--r--. 1 42834 majoroslab  33879 Mar  6 21:08 GRCh38_HBB.tsv
-rw-r--r--. 1 42834 majoroslab  49575 Mar  6 21:56 GRCh38_HBG1.tsv
-rw-r--r--. 1 42834 majoroslab  54775 Mar  6 21:57 GRCh38_HNF4A.tsv
-rw-r--r--. 1 42834 majoroslab  81091 Mar  6 21:57 GRCh38_IRF4.tsv
-rw-r--r--. 1 42834 majoroslab 105057 Mar  6 21:57 GRCh38_IRF6.tsv
-rw-r--r--. 1 42834 majoroslab  59475 Mar  6 21:58 GRCh38_LDLR.2.tsv
-rw-r--r--. 1 42834 majoroslab  59044 Mar  6 21:58 GRCh38_LDLR.tsv
-rw-r--r--. 1 42834 majoroslab 113536 Mar  6 21:59 GRCh38_MSMB.tsv
-rw-r--r--. 1 42834 majoroslab  89358 Mar  6 21:59 GRCh

**Files are also spliteted by promoters and enhancers**

In [4]:
ls /hpc/group/igvf/benchmarks/Kircher_et_al_2019/Promoters

GRCh38_F9.tsv	  GRCh38_HBG1.tsv    GRCh38_MSMB.tsv	  GRCh38_TERT-GBM.tsv
GRCh38_FOXE1.tsv  GRCh38_HNF4A.tsv   GRCh38_PKLR-24h.tsv  GRCh38_TERT-GSc.tsv
GRCh38_GP1BA.tsv  GRCh38_LDLR.2.tsv  GRCh38_PKLR-48h.tsv  GRCh38_TERT-HEK.tsv
GRCh38_HBB.tsv	  GRCh38_LDLR.tsv    GRCh38_TERT-GAa.tsv


In [5]:
ls /hpc/group/igvf/benchmarks/Kircher_et_al_2019/Enhancers

GRCh38_BCL11A.tsv	  GRCh38_RET.tsv	 GRCh38_UC88.tsv
GRCh38_IRF4.tsv		  GRCh38_SORT1.2.tsv	 GRCh38_ZFAND3.tsv
GRCh38_IRF6.tsv		  GRCh38_SORT1-flip.tsv  GRCh38_ZRSh-13h2.tsv
GRCh38_MYCrs11986220.tsv  GRCh38_SORT1.tsv	 GRCh38_ZRSh-13.tsv
GRCh38_MYCrs6983267.tsv   GRCh38_TCF7L2.tsv


**Promoters.txt and Enhancers.txt are tables from https://kircherlab.bihealth.org/satMutMPRA/**

In [6]:
cat /hpc/group/igvf/benchmarks/Kircher_et_al_2019/Promoters.txt | head -n 3

Name	Genomic coordinates (GRCh37)	Genomic coordinates (GRCh38)	Transcript	Associated Phenotype	Luciferase vector	MPRA vector	Cell line	Transf. time (hr)	Fold change (Wild type)	Fold Change (MPRA)	Construct size (bp)
F9	"chrX:138,612,622-138,612,924"	"chrX:139,530,463-139,530,765"	NM_000133.3	Hemophilia B	pGL4.11b	pGL4.11c	HepG2	24	2.6	2.1	303
FOXE1	"chr9:100,615,537-100,616,136"	"chr9:97,853,255-97,853,854"	NM_004473.3	Thyroid cancer	pGL4.11b	pGL4.11c	HeLa	24	4.2	2.5	600


In [7]:
cat /hpc/group/igvf/benchmarks/Kircher_et_al_2019/Enhancers.txt | head -n 3

Name	Genomic coordinates (GRCh37)	Genomic coordinates (GRCh38)	Associated Phenotype	Luciferase vector	MPRA vector	Cell line	Transf. time (hr)	Fold change (Wild type)	Fold Change (MPRA)	Construct size (bp)
BCL11A+58	"chr2:60,722,075-60,722,674"	"chr2:60,494,940-60,495,539"	Sickle cell disease	pGL4.23	pGL4.23d	HEL 92.1.7	24	2.5	1.7	600
IRF4	"chr6:396,143-396,593"	"chr6:396,143-396,593"	Human pigmentation	pGL4.23	pGL4.23d	SK-MEL-28	24	44.5	16.3	451


## Kircher data is used for BlueSTARR model evaluation

In [8]:
ls /hpc/group/igvf/K562

combined.input_and_output.gt_100.log2FC.sequence.txt.gz
combined.input_and_output.gt_150.log2FC.sequence.txt.gz
combined.input_and_output.gt_200.log2FC.sequence.txt.gz
CRE-preds
full-set
K562.combined.input_and_output.w600s50.gt_200.log2FC.sequence.txt.gz
K562.combined.input_and_output.w600s50.gt_200.thres_dna_only.log2FC.sequence.txt.gz
leave-one-out
mdl_eval
README.md


In [9]:
ls /hpc/group/igvf/K562/mdl_eval

BlueSTARR_vs_DeltaSVM  slurm-biased  slurm-unbiased


In [10]:
ls /hpc/group/igvf/K562/mdl_eval/BlueSTARR_vs_DeltaSVM

AUC-ROC.ipynb			    get_Kircher_chunks.ipynb
blue_bird_delta_enh_counts.csv.gz   get_Kircher_pred.py
blue_bird_delta_enh.csv.gz	    Kircher_entire_regions
blue_bird_delta_prom_counts.csv.gz  Kircher_infilled_regions
blue_bird_delta_prom.csv.gz	    parquets_K562_bluestarr
blue_kircher_delta_enh.csv.gz	    parquets_K562_enh
blue_kirchir_delta.csv		    parquets_K562_prom
BlueSTARR_vs_Kirchir.ipynb


In [11]:
TXT_FDIRY=/hpc/group/igvf/K562/mdl_eval/BlueSTARR_vs_DeltaSVM/Kircher_entire_regions
ls ${TXT_FDIRY}

Kircher_region_chunk.txt      Kircher_region_pred.out
Kircher_region.coords	      Kircher_region_pred_processed_DMSO.txt
Kircher_region.fasta	      Kircher_region_pred_processed.txt
Kircher_region_pred_DMSO.txt  Kircher_region_pred.sh
Kircher_region_pred.err       Kircher_region_pred.txt


**Region chunk**

In [12]:
TXT_FNAME=Kircher_region_chunk.txt
TXT_FPATH=${TXT_FDIRY}/${TXT_FNAME}

ls ${TXT_FPATH}

/hpc/group/igvf/K562/mdl_eval/BlueSTARR_vs_DeltaSVM/Kircher_entire_regions/Kircher_region_chunk.txt


**Region coordinate**

In [13]:
TXT_FNAME=Kircher_region.coords
TXT_FPATH=${TXT_FDIRY}/${TXT_FNAME}

ls   ${TXT_FPATH}
echo 
head ${TXT_FPATH}

/hpc/group/igvf/K562/mdl_eval/BlueSTARR_vs_DeltaSVM/Kircher_entire_regions/Kircher_region.coords

chr11:5226964-5227264
chr11:5249791-5250091
chr20:44355511-44355811
chr5:1294967-1295267
chr8:127519269-127519569
chr8:127519432-127519732
chr10:112998239-112998539
chr10:112998389-112998689
chr10:112998539-112998839
chrX:139530462-139530762


In [14]:
cat ${TXT_FPATH} | grep chr11:5227021-5227208

: 1

**Region fasta**

In [38]:
TXT_FNAME=Kircher_region.fasta
TXT_FPATH=${TXT_FDIRY}/${TXT_FNAME}

ls   ${TXT_FPATH}
echo 
head ${TXT_FPATH}

/hpc/group/igvf/K562/mdl_eval/BlueSTARR_vs_DeltaSVM/Kircher_entire_regions/Kircher_region.fasta

>chr11:5226964-5227264
CACCTTGCCCCACAGGGCAGTAACGGCAGACTTCTCCTCAGGAGTCAGAT
GCACCATGGTGTCTGTTTGAGGTTGCTAGTGAACACAGTTGTGTCAGAAG
CAAATGTAAGCAATAGATGGCTCTGCCCTGACTTTTATGCCCAGCCCTGG
CTCCTGCCCTCCCTGCTCCTGGGAGTAGATTGGCCAACCCTAGGGTGTGG
CTCCACAGGGTGAGGTCTAAGTGATGACAGCCGTACCTGTCCTTGGCTCT
TCTGGCACTGGCTTAGGAGTTGGACTTCAAACCCTCAGCCCTCCCTCTAA
>chr11:5249791-5250091
TGAAATGACCCATGGCGTCTGGACTAGGAGCTTATTGATAACCTCAGACG
TTCCAGAAGCGAGTGTGTGGAACTGCTGAAGGGTGCTTCCTTTTATTCTT


**Region effect**

In [37]:
TXT_FNAME=Kircher_region_pred_processed.txt
TXT_FPATH=${TXT_FDIRY}/${TXT_FNAME}

ls   ${TXT_FPATH}
echo 
head ${TXT_FPATH}

/hpc/group/igvf/K562/mdl_eval/BlueSTARR_vs_DeltaSVM/Kircher_entire_regions/Kircher_region_pred_processed.txt

chrom	Kircher_start	Kircher_end	effect	SPDI
chr11	5227021	5227208	0.0049742682314810605	NC_000011.10:5227021:G:C
chr11	5227021	5227208	-0.025114175586687687	NC_000011.10:5227021:G:A
chr11	5227021	5227208	-0.007699144063009644	NC_000011.10:5227021:G:T
chr11	5227021	5227208	-0.014522990617761605	NC_000011.10:5227022:G:C
chr11	5227021	5227208	0.004607564005987913	NC_000011.10:5227022:G:A
chr11	5227021	5227208	0.024201122749209936	NC_000011.10:5227022:G:T
chr11	5227021	5227208	0.007836964720265821	NC_000011.10:5227023:T:G
chr11	5227021	5227208	0.09841981171284618	NC_000011.10:5227023:T:C
chr11	5227021	5227208	0.05130660701998546	NC_000011.10:5227023:T:A


## Notebook: Get Kircher chunks

The four files above are generated using the following notebook:

`/hpc/group/igvf/K562/mdl_eval/BlueSTARR_vs_DeltaSVM/get_Kircher_chunks.ipynb`

This notebook shows how Bill center the 300 bp over the regions and mutate bases within the region.

- Step 1. import each region and length
    - Example: `chr8:127519269-127519732 463`
    - Note: In the table: `chr8:127,519,270-127,519,732 464`
- Step 2. 

```
# Kircher region: 21 = unmatched + full (without cutting)
# unmatched:  5
# pre cre:  7
# post cre:  8
# inter cre:  11
# full cre:  16
# all missing:  26 = pre + post + inter
# Kirchir region cut by cCRE: 31 = unmatched + all_missing
```

## Prediction results

In [6]:
head /hpc/group/igvf/K562/mdl_eval/BlueSTARR_vs_DeltaSVM/Kircher_entire_regions/Kircher_region_pred.txt

chr11:5227021-5227208	chr11:5226964-5227264	pos=5227021	ref=G	G	-0.1384411
chr11:5227021-5227208	chr11:5226964-5227264	pos=5227021	ref=G	C	-0.1349932
chr11:5227021-5227208	chr11:5226964-5227264	pos=5227021	ref=G	A	-0.15584892
chr11:5227021-5227208	chr11:5226964-5227264	pos=5227021	ref=G	T	-0.14377774
chr11:5227021-5227208	chr11:5226964-5227264	pos=5227022	ref=G	G	-0.1384411
chr11:5227021-5227208	chr11:5226964-5227264	pos=5227022	ref=G	C	-0.14850767
chr11:5227021-5227208	chr11:5226964-5227264	pos=5227022	ref=G	A	-0.13524738
chr11:5227021-5227208	chr11:5226964-5227264	pos=5227022	ref=G	T	-0.12166616
chr11:5227021-5227208	chr11:5226964-5227264	pos=5227023	ref=T	G	-0.13300893
chr11:5227021-5227208	chr11:5226964-5227264	pos=5227023	ref=T	C	-0.070221685
